### PR TITLE
[CI] Fix some core shuffle tests 

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
Three core shuffle release tests are failing because i3 machine is deprecated.

The i3 machine is deprecated.  Change machines type to m6i. The difference between the two type machine is that i3 uses locally SSD while m6i uses EPS through network bandwidth. EPS seems to work just fine.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests
   -  https://buildkite.com/ray-project/release-tests-pr/builds/36856
   - https://buildkite.com/ray-project/release-tests-pr/builds/36867 (failed tests are currently failing on master)

Performance data:

**shuffle_20gb_with_state_api.aws**
```
NEW

time = 129.80361704
--
  | success = 1
  | perf_metrics = [{'perf_metric_name': 'state_api_extra_latency_sec', 'perf_metric_value': 36.065432741999985, 'perf_metric_type': 'LATENCY'}, {'perf_metric_name': 'state_api_extra_latency_sec_percentage', 'perf_metric_value': 78.68508210428904, 'perf_metric_type': 'LATENCY'}, {'perf_metric_name': 'state_api_extra_mem', 'perf_metric_value': -0.18999999999999995, 'perf_metric_type': 'MEMORY'}]

OLD

time = 59.80607887200001
--
  | success = 1
  | perf_metrics = [{'perf_metric_name': 'state_api_extra_latency_sec', 'perf_metric_value': 27.260793666000005, 'perf_metric_type': 'LATENCY'}, {'perf_metric_name': 'state_api_extra_latency_sec_percentage', 'perf_metric_value': 181.9177483926357, 'perf_metric_type': 'LATENCY'}, {'perf_metric_name': 'state_api_extra_mem', 'perf_metric_value': -0.06000000000000005, 'perf_metric_type': 'MEMORY'}]
```

**shuffle_100gb.aws**
```
NEW

shuffle_time = 84.73047876358032
--
  | success = 1
  | perf_metrics = [{'perf_metric_name': 'shuffle_time', 'perf_metric_value': 84.73047876358032, 'perf_metric_type': 'LATENCY'}]

OLD

shuffle_time = 85.13335418701172
--
  | success = 1
  | perf_metrics = [{'perf_metric_name': 'shuffle_time', 'perf_metric_value': 85.13335418701172, 'perf_metric_type': 'LATENCY'}]
```

**autoscaling_shuffle_1tb_1000_partitions.aws**

```
NEW

shuffle_time = 708.3939363956451
--
  | success = 1
  | perf_metrics = [{'perf_metric_name': 'shuffle_time', 'perf_metric_value': 708.3939363956451, 'perf_metric_type': 'LATENCY'}]

OLD

shuffle_time = 742.8974359035492
--
  | success = 1
  | perf_metrics = [{'perf_metric_name': 'shuffle_time', 'perf_metric_value': 742.8974359035492, 'perf_metric_type': 'LATENCY'}]
```

**dask_on_ray_100gb_sort.aws**
```
NEW
duration = 4110.363643169403
--
  | success = 1

OLD

duration = 2549.20721244812
--
  | success = 1
```

** dask_on_ray_1tb_sort**
```
NEW
duration = 5466.110460519791
--
  | success = 1

OLD


duration = 5006.332159757614
--
  | success = 1

```

** chaos_dask_on_ray_large_scale_test_no_spilling.aws**

```
NEW
runtime: 722.30
success = 1
--
  | _peak_memory = 24.11

OLD
runtime: 1233.42
success = 1
--
  | _peak_memory = 26.56

```

### CODE REVIEW

**Review by rkooo567 — COMMENTED** [review]
can you make sure the memory capacity of m5 is equivalent to i3? Also what does it mean by "i3 is deprecated"?

**Comment by can-anyscale** [comment]
@rkooo567 : for context https://anyscaleteam.slack.com/archives/C01R4DJV3B5/p1682982857441099

File: release/nightly_tests/dask_on_ray/1tb_sort_compute.yaml [hunk 1]
```diff
@@ -10,13 +10,13 @@ aws:
 
 head_node_type:
     name: head_node
-    instance_type: i3.8xlarge
+    instance_type: m5.8xlarge
```
Comment (aslonnie) [inline release/nightly_tests/dask_on_ray/1tb_sort_compute.yaml:13]: why not change to `m6i`? which is cheaper and faster?

File: release/nightly_tests/shuffle/shuffle_compute_multi.yaml [hunk 2]
```diff
@@ -12,12 +12,12 @@ aws:
 
 head_node_type:
     name: head_node
-    instance_type: i3.4xlarge
+    instance_type: m5.4xlarge # i3.4xlarge
```
Comment (aslonnie) [inline release/nightly_tests/shuffle/shuffle_compute_multi.yaml:15]: what is this comment for?

File: release/nightly_tests/dask_on_ray/1tb_sort_compute.yaml (hunk 1 above)
Comment (can-anyscale) [reply release/nightly_tests/dask_on_ray/1tb_sort_compute.yaml:13]: wow cheaper and faster exists?

**Comment by jjyao** [comment]
Could you update the PR description, it still mentions `m5.4xlarge`.

Also m6i has less memory, does it affect the runtime?

**Comment by can-anyscale** [comment]
In theory it should, not because of memory but because network bandwidth as well, so we'll need to recompute the performance data with this new spec going forward

**Comment by can-anyscale** [comment]
@jjyao: if you think we MUST have that i3 machines to keep things the same, I guess we can have that conversation with product as well

**Comment by jjyao** [comment]
@can-anyscale

> In theory it should, not only because of memory but because network bandwidth as well

Could you paste the runtime before and after so we know the diff?

**Comment by can-anyscale** [comment]
@jjyao I pasted a few in the PR summary so you can get some ideas

**Review by jjyao — APPROVED** [review]

**Pull request merged**